python/tests/transfer_trigger_tests.py

```diff
@@ -210,3 +216,1349 @@ def test_trigger_manager_remove(self):
+        def after_read(dataframe, new, old, current):
+            # Need to use this type of assert because this is executes outside this class
+            assert (str(type(dataframe)) ==
+                    "<class 'rtypes.dataframe.dataframe.dataframe'>")
+            assert (new == None)
+            assert (str(type(old)) ==
+                    "<class 'tests.trigger_test_classes.Transaction'>")
+            assert (str(type(current)) ==
+                    "<class 'tests.trigger_test_classes.Transaction'>")
+
+        DF = dataframe()
+        DF.add_trigger(after_read)
+        DF.add_type(Transaction)
+        DF.append(Transaction, Transaction("Robert", 10))
+        DF.get(Transaction)
+
+    def test_ar_2(self):
+        """
+        Tests that rasing Block Action in after read triggers does nothing
+        """
+        @trigger(Transaction, TriggerTime.after, TriggerAction.read)
+        def after_read(datatframe, new, old, current):
+            if old.name == "Robert":
+                raise BlockAction
+
+        DF = dataframe()
+        DF.add_trigger(after_read)
+        DF.add_type(Transaction)
+        test_1 = Transaction("Robert", 10)
+        DF.append(Transaction, test_1)
+        self.assertEqual(DF.get(Transaction, "Robert"), test_1)
+
+    def test_ar_3(self):
+        """
+        After read triggers cannot alter demensions of an object that is being CRUD
+        """
+        @trigger(Transaction, TriggerTime.after, TriggerAction.read)
+        def after_read(datatframe, new, old, current):
+            old.amount = 100
+            current.amount = 100
+
+        DF = dataframe()
+        DF.add_trigger(after_read)
+        DF.add_type(Transaction)
+        test = Transaction("Robert", 10)
+        DF.append(Transaction, test)
+        self.assertTrue(DF.get(Transaction, "Robert") == test)
+
+    # Cannot read data of type x within a trigger attached to type x
+    '''
+    def test_ar_4(self):
+        """
+        Tests that after read triggers can [access|change]
+        objects in dataframe of the [same] type
+        """
+        @trigger(Transaction, TriggerTime.after, TriggerAction.read)
+        def after_read(dataframe, new, old, current):
+            jake = dataframe.get(Transaction, "Jake")
+            jake.amount = 1000
+
+        DF = dataframe()
+        DF.add_trigger(after_read)
+        DF.add_type(Transaction)
+        DF.append(Transaction, Transaction("Jake", 10))
+        DF.append(Transaction, Transaction("Robert", 10))
+        print(DF.get(Transaction))
+    '''
+
+    def test_ar_5(self):
+        """
+        Tests that after create triggers can [access|change]
+        objects in dataframe of a [different] type
+        """
+        @trigger(Transaction, TriggerTime.after, TriggerAction.read)
+        def after_read(dataframe, new, old, current):
+            log = dataframe.get(TransactionHistory, "Robert")
+            log.count += 1
+
+        DF = dataframe()
```

Comment:
use lower case df for all dataframe objects.

General convention: if the variable is not a global variable, use lower case (and this_style_casing)

Reply:
Change it everywhere.